drivers/pds/lib/resiliency_utils.go

```diff
@@ -0,0 +1,174 @@
+package lib
+
+import (
+	"errors"
+	"sync"
+
+	pds "github.com/portworx/pds-api-go-client/pds/v1alpha1"
+	"github.com/portworx/torpedo/drivers/node"
+
+	_ "github.com/portworx/torpedo/drivers/scheduler/dcos"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/portworx/torpedo/pkg/log"
+	"github.com/portworx/torpedo/tests"
+)
+
+const (
+	ActiveNodeRebootDuringDeployment = "active-node-reboot-during-deployment"
+)
+
+// PDS vars
+var (
+	wg                        sync.WaitGroup
+	ResiliencyFlag            = false
+	hasResiliencyConditionMet = false
+	FailureType               TypeOfFailure
+	testError                 error
+	conditionError            error
+	checkTillReplica          int32
+	ResiliencyCondition       = make(chan bool)
+)
+
+// Struct Definition for kind of Failure the framework needs to trigger
+type TypeOfFailure struct {
+	Type   string
+	Method func() error
+}
+
+// Wrapper to Define failure type from Test Case
+func DefineFailureType(failuretype TypeOfFailure) {
+	FailureType = failuretype
+}
+
+// Executes all methods in parallel
+func ExecuteInParallel(functions ...func()) {
+	wg.Add(len(functions))
+	defer wg.Wait()
+	for _, fn := range functions {
+		go func(FuncToRun func()) {
+			defer wg.Done()
+			FuncToRun()
+		}(fn)
+	}
+}
+
+// Function to enable Resiliency Test
+func MarkResiliencyTC(resiliency bool) {
+	ResiliencyFlag = resiliency
+	if resiliency {
+		tests.InitInstance()
+	}
+}
+
+// Function to wait for event to induce failure
+func InduceFailure(failure string, ns string) {
+	isResiliencyConditionset := <-ResiliencyCondition
+	if isResiliencyConditionset {
+		FailureType.Method()
+	} else {
+		testError = errors.New("Resiliency Condition did not meet. Failing this test case.")
+		return
+	}
+	return
+}
+
+// Close all open Resiliency channels here
+func CloseResiliencyChannel() {
+	// Close the Channel if it's empty. Otherwise there is no need to close as per Golang official documentation,
+	// as far as we are making sure no writes are happening to a closed channel. Make sure to call this method only
+	// during Post Test Case execution to avoid any unknown panics
+	if len(ResiliencyCondition) == 0 {
+		close(ResiliencyCondition)
+	}
+}
+
+//
+func InduceFailureAfterWaitingForCondition(deployment *pds.ModelsDeployment, namespace string, CheckTillReplica int32) error {
+	switch FailureType.Type {
+	// Case when we want to reboot a node onto which a deployment pod is coming up
+	case ActiveNodeRebootDuringDeployment:
+		checkTillReplica = CheckTillReplica
+		log.InfoD("Entering to check if Data service has %v active pods. Once it does, we will reboot the node it is hosted upon.", checkTillReplica)
+		func1 := func() {
+			GetPdsSs(deployment.GetClusterResourceName(), namespace, checkTillReplica)
```

Comment:
how are you handling the error returned here?

Reply:
It's getting handled as a separate error check condition once the go routine exits.